### PR TITLE
Edited article for CDS

### PR DESCRIPTION
**Pull request by panissidi**

### CODE REVIEW

File: content/tutorials/webrtc/datachannels/en/index.html [hunk 1]
```diff
@@ -71,79 +71,78 @@ <h2 id="why-another-data-channel">Why another data channel?</h2>
...
+<h2>A caveat: Signaling, STUN, and TURN</h2>
 
-<p>Although WebRTC enables peer-to-peer communication, it still needs servers:</p>
+<p>Although WebRTC enables peer-to-peer communication, it still needs servers for:</p>
 
 <ul>
-  <li><strong>For signaling:</strong> to enable the exchange of media and network metadata to bootstrap a peer connection.</li>
-  <li><strong>To cope with NATs and firewalls</strong>: by using the ICE framework to establish the best possible network path between peers, by working with STUN servers (to ascertain a publicly accessible IP and port for each peer) and TURN servers (if direct connection fails and data relaying is required).</li>
+  <li>Signaling&mdash;the exchange of media and network metadata to bootstrap a peer connection</li>
+  <li>Coping with network address translators (NATs) and firewalls</li>
```
Comment (samdutton) [inline content/tutorials/webrtc/datachannels/en/index.html:89]: I'd be inclined to leave in the rest of this sentence, or including the information some other ways. 

Maybe these items could become the bullet points, something like this:

```
WebRTC enables peer-to-peer communication, but it still needs servers for <strong>signaling</strong>, to exchange media and network metadata to bootstrap a peer connection. 

WebRTC copes with NATs and firewalls by using: 
<ul>
  <li><strong>The ICE framework</strong> to establish the best possible network path between peers.</li>
  <li><strong>STUN servers</strong> to ascertain a publicly accessible IP and port for each peer</li>
  <li><strong>TURN servers</strong> if direct connection fails and data relaying is required.</li>	 
```

File: content/tutorials/webrtc/datachannels/en/index.html [hunk 2]
```diff
@@ -57,93 +57,92 @@
...
-  <li><strong>Unreliable and unordered mode</strong> does not guarantee every message will get to the other side nor what order they get there. This removes the overhead, allowing this mode to work much faster.</li>
-  <li><strong>Partial reliable mode</strong> guarantees the transmission of message under a specific condition (such as a retransmit timeout or a maximum amount of retransmissions). The ordering of messages is also configurable.
-</ul>
-
-<p>Performance for the first two modes is about the same when there are no packet losses. However, in reliable and ordered mode a lost packet will cause other packets to get blocked behind it, and the lost packet might be stale by the time it is retransmitted and arrives. It is, of course, possible to use multiple data channels within the same application, each with their own (un)reliable semantics.</p>
-
-<p>We show below how to configure RTCDataChannel to use reliable and ordered or unreliable and unordered mode.</p>
+<p>Next, you learn how to configure <code>RTCDataChannel</code> to use reliable and ordered or unreliable and unordered mode.</p>
```
Comment (samdutton) [inline content/tutorials/webrtc/datachannels/en/index.html:132]: The table is adapted from Ilya's book (i.e. needs the credit).

File: content/tutorials/webrtc/datachannels/en/index.html [hunk 3]
```diff
@@ -57,93 +57,92 @@
...
 </tr>
 <tr>
-  <td style="font-weight: bold;">Congestion control</td><td>yes</td><td>no</td><td>yes</td>
+  <td style="font-weight: bold;">Congestion control</td><td>Yes</td><td>No</td><td>Yes</td>
 </tr>
 </table>
 
-<p>(From <a href="http://chimera.labs.oreilly.com/books/1230000000545/ch18.html">High Performance Browser Networking</a> by <a href="http://www.igvita.com/" title="Ilya Grigorik's website">Ilya Grigorik</a>.)</p>
-
-<p>RTCDataChannel can work in unreliable and unordered mode (analogous to User Datagram Protocol or UDP), reliable and ordered mode (analogous to Transmission Control Protocol or TCP) and partial reliable modes:</p>
```
Comment (samdutton) [inline content/tutorials/webrtc/datachannels/en/index.html:122]: These deleted paragraphs should be reinstated :).

File: content/tutorials/webrtc/datachannels/en/index.html [hunk 4]
```diff
@@ -57,93 +57,92 @@
...
-  <li><a href="http://webrtc.github.io/samples/src/content/datachannel/basic/">Transmit text</a></li>
-  <li><a href="http://webrtc.github.io/samples/src/content/datachannel/filetransfer/">Transfer a file</a></li>
-  <li><a href="http://pubnub.github.io/webrtc">pubnub.github.io/webrtc</a> (between two PubNub clients)</li>
+ <li><a href="https://simpl.info/dc">simpl.info <code>RTCDataChannel</code></a></li>
+  <li><a href="https://webrtc.github.io/samples/src/content/datachannel/basic/">WebRTC samples Transmit text</a></li>
+  <li><a href="https://webrtc.github.io/samples/src/content/datachannel/filetransfer/">WebRTC samples Transfer a file</a></li>
 </ul>
 
-<p>In these examples the browser is making a peer connection to itself. It is then creating a data channel and sending the message along the peer connection. Finally, your message appears in the box on the other side of the page!</p>
+<p>In these examples, the browser is making a peer connection to itself. It is then creating a data channel and sending the message along the peer connection. Finally, your message appears in the box on the other side of the page!</p>
```
Comment (samdutton) [inline content/tutorials/webrtc/datachannels/en/index.html:145]: Maybe reword slightly: 

In these examples, the browser makes a peer connection to itself, then creates a data channel and sends a message via the peer connection.

File: content/tutorials/webrtc/datachannels/en/index.html [hunk 5]
```diff
@@ -180,46 +179,46 @@ <h2 id="just-show-me-the-action">Configuring data channels</h2>
 };
 </pre>
 
-<p>It is also possible to add a <code>maxRetransmits</code> option (the number of times to try before failing) but you can only specify maxRetransmits or maxPacketLifeTime, not both. For UDP semantics, set <code>maxRetransmits</code> to 0 and <code>ordered</code> to <code>false</code>. For more information, see <a href="http://tools.ietf.org/html/rfc4960" title="IETF RFC 4960">RFC 4960</a> (SCTP) and <a href="http://tools.ietf.org/html/rfc3758" title="IETF RFC 3758">RFC 3758</a> (SCTP partial reliability).</p>
+<p>It is also possible to add a <code>maxRetransmits</code> option (the number of times to try before failing), but you can only specify maxRetransmits or maxPacketLifeTime, not both. For UDP semantics, set <code>maxRetransmits</code> to <code>0</code> and <code>ordered</code> to <code>false</code>. For more information, see <a href="https://tools.ietf.org/html/rfc4960" title="IETF RFC 4960">RFC 4960 - SCTP</a> and <a href="https://tools.ietf.org/html/rfc3758" title="IETF RFC 3758">RFC 3758 - SCTP Partial Reliability Extension</a>.</p>
```
Comment (samdutton) [inline content/tutorials/webrtc/datachannels/en/index.html:182]: I think it might be better just to put the names in full and remove the RFC numbers, something like this:

```
For more information, see the following IETF RFCs: <a href="https://tools.ietf.org/html/rfc4960" title="IETF RFC 4960">Stream Control Transmission Protocol</a> and <a href="https://tools.ietf.org/html/rfc3758" title="IETF RFC 3758">Stream Control Transmission Protocol Partial Reliability Extension</a>.
```

File: content/tutorials/webrtc/datachannels/en/index.html [hunk 6]
```diff
@@ -180,46 +179,46 @@ <h2 id="just-show-me-the-action">Configuring data channels</h2>
...
 
 <h2 id="is-it-safe">Is it safe?</h2>
 
-<p>Encryption is mandatory for all WebRTC components. With RTCDataChannel all data is secured with <a href="https://en.wikipedia.org/wiki/Datagram_Transport_Layer_Security" title="Wikipedia article about DTLS">Datagram Transport Layer Security</a> (DTLS). DTLS is a derivative of SSL, meaning your data will be as secure as using any standard SSL based connection. DTLS is standardized and built in to all browsers that support WebRTC. You can find more information about DTLS on the <a href="http://wiki.wireshark.org/DTLS">Wireshark wiki</a>.</p>
+<p>Encryption is mandatory for all WebRTC components. With <code>RTCDataChannel</code>, all data is secured with <a href="https://en.wikipedia.org/wiki/Datagram_Transport_Layer_Security" title="Wikipedia article about DTLS">Datagram Transport Layer Security (DTLS)</a>. DTLS is a derivative of SSL, meaning your data will be as secure as using any standard SSL-based connection. DTLS is standardized and built into all browsers that support WebRTC. For more information, see <a href="https://wiki.wireshark.org/DTLS">DTLS - The Wireshark wiki</a>.</p>
```
Comment (samdutton) [inline content/tutorials/webrtc/datachannels/en/index.html:199]: I think DTLS should probably be left out of the link text, since it's not in the article title.

Comment (samdutton) [reply content/tutorials/webrtc/datachannels/en/index.html:199]: Similar for the Wireshark link:

```
For more information, see the <a href="https://gitlab.com/wireshark/wireshark/-/wikis/DTLS">Wireshark Wiki DTLS article</a>.
```

**Comment by samdutton** [comment]
Good edits! Added some comments and did a couple of commits to update links to HTTPS and fix a few minor glitches.

File: content/tutorials/webrtc/datachannels/en/index.html (hunk 1 above)
Comment (panissidi) [reply content/tutorials/webrtc/datachannels/en/index.html:89]: This copy is great! Thanks!

I also added links, but please check them.

File: content/tutorials/webrtc/datachannels/en/index.html (hunk 3 above)
Comment (panissidi) [reply content/tutorials/webrtc/datachannels/en/index.html:122]: These are still in the article, but I moved them a little higher.

File: content/tutorials/webrtc/datachannels/en/index.html (hunk 2 above)
Comment (panissidi) [reply content/tutorials/webrtc/datachannels/en/index.html:132]: The attribution is still there, just before the table.

File: content/tutorials/webrtc/datachannels/en/index.html (hunk 4 above)
Comment (panissidi) [reply content/tutorials/webrtc/datachannels/en/index.html:145]: Will fix

File: content/tutorials/webrtc/datachannels/en/index.html (hunk 5 above)
Comment (panissidi) [reply content/tutorials/webrtc/datachannels/en/index.html:182]: Will fix

File: content/tutorials/webrtc/datachannels/en/index.html (hunk 6 above)
Comment (panissidi) [reply content/tutorials/webrtc/datachannels/en/index.html:199]: Will fix

**Comment by samdutton** [comment]
Resolved conflicts (and updated broken link to TURN servers).
